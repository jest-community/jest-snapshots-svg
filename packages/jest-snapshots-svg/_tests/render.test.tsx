import "../index"

import * as React from "react"
import { View } from "react-native"
import * as renderer from "react-test-renderer"

import componentTreeToNodeTree from "../component-tree-to-nodes"
import renderedComponentTree from "../reapply-layouts-to-components"
import treeToSVG from "../tree-to-svg"

import * as fs from "fs"
import * as yoga from "yoga-layout"

it("handles some simple JSX", () => {
  const jsx = (
    <View style={{width: 100, height: 200, marginLeft: 40, marginTop: 40, paddingLeft: 20}}>
      <View style={{width: 20, height: 20}}/>
      <View style={{width: 160, height: 40, marginTop: 20}}>
        <View style={{width: 10, height: 10, marginLeft: 10, marginTop: 20}}/>
        <View style={{width: 10, height: 10, marginLeft: 20, marginTop: 10}}/>
      </View>
      <View style={{width: 20, height: 20}}/>
      <View style={{width: 160, height: 40, marginTop: 20, paddingHorizontal: 10 }}>
        <View style={{width: 10, height: 10, marginHorizontal: 10 }}/>
        <View style={{width: 10, height: 10, marginVertical: 10 }}/>
        <View style={{width: 10, height: 10, marginHorizontal: -10 }}/>
      </View>
    </View>
  )

  const component = renderer.create(jsx).toJSON()
  const settings = {
    width:  600,
    height: 400,
  }

  const rootNode = componentTreeToNodeTree(component, settings)
  const rendered = renderedComponentTree(component, rootNode)
  const results = treeToSVG(rendered, settings)
  expect(results).toMatchSnapshot()

  expect(component).toMatchSVGSnapshot(1024, 768)
  rootNode.freeRecursive()
})
