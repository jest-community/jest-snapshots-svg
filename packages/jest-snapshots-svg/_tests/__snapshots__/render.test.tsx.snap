// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles some simple JSX 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>
<svg width=\\"600\\" height=\\"400\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\">

 <rect type=\\"View\\" fill-opacity=\\"0.1\\" stroke-width=\\"1\\" stroke=\\"black\\" x=\\"0\\" y=\\"0\\" width=\\"NaN\\" height=\\"NaN\\"/>

 <g transform='translate(0, 0)'>
   <rect type=\\"View\\" fill-opacity=\\"0.1\\" stroke-width=\\"1\\" stroke=\\"black\\" x=\\"0\\" y=\\"0\\" width=\\"NaN\\" height=\\"NaN\\"/>
   <rect type=\\"View\\" fill-opacity=\\"0.1\\" stroke-width=\\"1\\" stroke=\\"black\\" x=\\"0\\" y=\\"0\\" width=\\"NaN\\" height=\\"NaN\\"/>

   <g transform='translate(0, 0)'>
     <rect type=\\"View\\" fill-opacity=\\"0.1\\" stroke-width=\\"1\\" stroke=\\"black\\" x=\\"0\\" y=\\"0\\" width=\\"NaN\\" height=\\"NaN\\"/>
     <rect type=\\"View\\" fill-opacity=\\"0.1\\" stroke-width=\\"1\\" stroke=\\"black\\" x=\\"0\\" y=\\"0\\" width=\\"NaN\\" height=\\"NaN\\"/>
   </g>

   <rect type=\\"View\\" fill-opacity=\\"0.1\\" stroke-width=\\"1\\" stroke=\\"black\\" x=\\"0\\" y=\\"0\\" width=\\"NaN\\" height=\\"NaN\\"/>
   <rect type=\\"View\\" fill-opacity=\\"0.1\\" stroke-width=\\"1\\" stroke=\\"black\\" x=\\"0\\" y=\\"0\\" width=\\"NaN\\" height=\\"NaN\\"/>

   <g transform='translate(0, 0)'>
     <rect type=\\"View\\" fill-opacity=\\"0.1\\" stroke-width=\\"1\\" stroke=\\"black\\" x=\\"0\\" y=\\"0\\" width=\\"NaN\\" height=\\"NaN\\"/>
     <rect type=\\"View\\" fill-opacity=\\"0.1\\" stroke-width=\\"1\\" stroke=\\"black\\" x=\\"0\\" y=\\"0\\" width=\\"NaN\\" height=\\"NaN\\"/>
     <rect type=\\"View\\" fill-opacity=\\"0.1\\" stroke-width=\\"1\\" stroke=\\"black\\" x=\\"0\\" y=\\"0\\" width=\\"NaN\\" height=\\"NaN\\"/>
   </g>

 </g>

</svg>
"
`;
